### ISSUE TITLE
`expect(clippy::macro_metavars_in_unsafe)` only works once

### ISSUE THREAD

**Issue by quuuuux** [rust-lang/rust-clippy#14488]
### Summary

If a macro containing `#[expect(clippy::macro_metavars_in_unsafe)]` is invoked multiple times then all invocations beyond the first will trigger an unfulfilled expectation warning.

### Lint Name

macro_metavars_in_unsafe

### Reproducer

I tried this code:

```rust
#[macro_export]
macro_rules! repro {
    ($x:expr) => {
        #[expect(clippy::macro_metavars_in_unsafe)]                                                                                                                                                                                                                                                                           
        unsafe { *(&$x as *const _) }
    }   
}

pub fn zero() -> i32 {
    repro!(0)
}

pub fn one() -> i32 {
    repro!(1)
}

```

I saw this happen:

```
warning: this lint expectation is unfulfilled
  --> src/lib.rs:4:18
   |
4  |         #[expect(clippy::macro_metavars_in_unsafe)]
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
14 |     repro!(1)
   |     --------- in this macro invocation
   |
   = note: `#[warn(unfulfilled_lint_expectations)]` on by default
   = note: this warning originates in the macro `repro` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen:

No warning.

### Version

```text
rustc 1.82.0 (f6e511eec 2024-10-15)
binary: rustc
commit-hash: f6e511eec7342f59a25f7c0534f1dbea00d01b14
commit-date: 2024-10-15
host: x86_64-unknown-linux-gnu
release: 1.82.0
LLVM version: 19.1.1
```

### Additional Labels

_No response_
